agent/config/config.go

```diff
@@ -179,122 +184,125 @@ type Config struct {
+	EnableLocalScriptChecks          *bool             `json:"enable_local_script_checks,omitempty" hcl:"enable_local_script_checks" mapstructure:"enable_local_script_checks"`
+	EnableSyslog                     *bool             `json:"enable_syslog,omitempty" hcl:"enable_syslog" mapstructure:"enable_syslog"`
+	EncryptKey                       *string           `json:"encrypt,omitempty" hcl:"encrypt" mapstructure:"encrypt"`
+	EncryptVerifyIncoming            *bool             `json:"encrypt_verify_incoming,omitempty" hcl:"encrypt_verify_incoming" mapstructure:"encrypt_verify_incoming"`
+	EncryptVerifyOutgoing            *bool             `json:"encrypt_verify_outgoing,omitempty" hcl:"encrypt_verify_outgoing" mapstructure:"encrypt_verify_outgoing"`
+	GossipLAN                        GossipLANConfig   `json:"gossip_lan,omitempty" hcl:"gossip_lan" mapstructure:"gossip_lan"`
+	GossipWAN                        GossipWANConfig   `json:"gossip_wan,omitempty" hcl:"gossip_wan" mapstructure:"gossip_wan"`
+	HTTPConfig                       HTTPConfig        `json:"http_config,omitempty" hcl:"http_config" mapstructure:"http_config"`
+	KeyFile                          *string           `json:"key_file,omitempty" hcl:"key_file" mapstructure:"key_file"`
+	LeaveOnTerm                      *bool             `json:"leave_on_terminate,omitempty" hcl:"leave_on_terminate" mapstructure:"leave_on_terminate"`
+	Limits                           Limits            `json:"limits,omitempty" hcl:"limits" mapstructure:"limits"`
+	LogLevel                         *string           `json:"log_level,omitempty" hcl:"log_level" mapstructure:"log_level"`
+	LogJSON                          *bool             `json:"log_json,omitempty" hcl:"log_json" mapstructure:"log_json"`
+	LogFile                          *string           `json:"log_file,omitempty" hcl:"log_file" mapstructure:"log_file"`
+	LogRotateDuration                *string           `json:"log_rotate_duration,omitempty" hcl:"log_rotate_duration" mapstructure:"log_rotate_duration"`
+	LogRotateBytes                   *int              `json:"log_rotate_bytes,omitempty" hcl:"log_rotate_bytes" mapstructure:"log_rotate_bytes"`
+	LogRotateMaxFiles                *int              `json:"log_rotate_max_files,omitempty" hcl:"log_rotate_max_files" mapstructure:"log_rotate_max_files"`
+	MaxQueryTime                     *string           `json:"max_query_time,omitempty" hcl:"max_query_time" mapstructure:"max_query_time"`
+	NodeID                           *string           `json:"node_id,omitempty" hcl:"node_id" mapstructure:"node_id"`
+	NodeMeta                         map[string]string `json:"node_meta,omitempty" hcl:"node_meta" mapstructure:"node_meta"`
+	NodeName                         *string           `json:"node_name,omitempty" hcl:"node_name" mapstructure:"node_name"`
+	// Enterprise Only
```

Comment:
nit: putting the comments at the end, wouldn't have changed this whole block.

Reply:
I moved all enterprise configurations to the end of the structs and it did reduce the diff quite a bit.